velox/vector/arrow/Bridge.cpp

```diff
@@ -790,7 +841,7 @@ void exportConstantValue(
         wrapInBufferViewAsViewer(vec.valuesAsVoid(), bufferSize),
         vec.mayHaveNulls() ? 1 : 0);
   }
-  exportToArrowImpl(*valuesVector, selection, out, pool);
+  exportToArrowImpl(*valuesVector, selection, out, pool, ArrowOptions());
```

Comment:
What if the input is a dictionary inside constant and we want to flatten the inner dictionary but not the outer constant?

Reply:
Why would a dictionary encoding be inside a constant encoding? As in, a single dictionary index and value that represents the constant? Would the parquet writer get such a data input?
This change assumes that either encoding wraps around flat data and not another encoding. 
It doesn't currently support this particular case.

The other unsupported case is when the dictionary/constant wraps a complex type.

Reply:
Can you add a check for these cases?

Reply:
The check for the scalar types is on line 795. If the wrapped type is not flat then return an error. Is that sufficient?

Reply:
You can add a check that `valueVector` is either null or flat

Reply:
Adding that constant vector cannot wrap another constant or dictionary at this time. There is a check that prohibits this scenario in the ConstantVector constructor